pootle/apps/pootle_app/management/commands/update_tmserver.py

```diff
@@ -0,0 +1,159 @@
+                body={
+                    'query': {
+                        'match_all': {}
+                    },
+                    'facets': {
+                        'stat1': {
+                            'statistical': {
+                                'field': 'revision'
+                            }
+                        }
+                    }
+                }
+            )
+            last_indexed_revision = result['facets']['stat1']['max']
+
+        self.stdout.write("Last indexed revision = %s" % last_indexed_revision)
+
+
+        units_qs = Unit.simple_objects \
+            .exclude(target_f__isnull=True) \
+            .exclude(target_f__exact='') \
+            .filter(revision__gt=last_indexed_revision) \
+            .select_related(
+                'submitted_by',
+                'store',
+                'store__translation_project__project',
+                'store__translation_project__language'
+            ).values(
+                'id',
+                'revision',
+                'source_f',
+                'target_f',
+                'submitted_by__username',
+                'submitted_by__full_name',
+                'submitted_by__email',
+                'store__translation_project__project__fullname',
+                'store__pootle_path',
+                'store__translation_project__language__code'
+            ).order_by()
+
+        total = units_qs.count()
+
+        if total == 0:
+            self.stdout.write("No translations to index")
+            sys.exit()
+
+        self.stdout.write("%s translations to index" % total)
+
+        if options['dry_run']:
+            sys.exit()
+
+        for i, unit in enumerate(units_qs.iterator()):
+            fullname = (unit['submitted_by__full_name'] or
+                        unit['submitted_by__username'])
+            project = unit['store__translation_project__project__fullname']
+
+            email_md5 = None
+            if unit['submitted_by__email']:
+                email_md5 = md5(unit['submitted_by__email']).hexdigest()
+
+            es.index(
+                index=INDEX_NAME,
+                doc_type=unit['store__translation_project__language__code'],
+                id=unit['id'],
+                body={
+                    'revision': int(unit['revision']),
+                    'project': project,
+                    'path': unit['store__pootle_path'],
+                    'username': unit['submitted_by__username'],
+                    'fullname': fullname,
+                    'email_md5': email_md5,
+                    'source': unit['source_f'],
+                    'target': unit['target_f'],
+                }
+            )
+
+            if (i % 1000 == 0) or (i == total):
+                percent = "%.1f" % (i * 100.0 / total)
+                self.stdout.write("%s (%s%%)" % (i, percent), ending='\r')
+                self.stdout.flush()
+
+        self.stdout.write("")
+
+        if i != total:
```

Comment:
Realize `enumerate()` provides a 0-based index, and the previous code was incrementing the index at the beginning of the loop, making that a 1-based index effectively. Therefore, here you need to `if i+1 != total:`, otherwise it'll always report it didn't load all of the expected translations.